flink-streaming-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraCommitter.java

```diff
@@ -0,0 +1,131 @@
+ */
+public class CassandraCommitter extends CheckpointCommitter {
+	private ClusterBuilder builder;
+	private transient Cluster cluster;
+	private transient Session session;
+
+	private String keySpace = "flink_auxiliary";
+	private String table = "checkpoints_";
+
+	private transient PreparedStatement deleteStatement;
+	private transient PreparedStatement updateStatement;
+	private transient PreparedStatement selectStatement;
+
+	public CassandraCommitter(ClusterBuilder builder) {
+		this.builder = builder;
+		ClosureCleaner.clean(builder, true);
+	}
+
+	public CassandraCommitter(ClusterBuilder builder, String keySpace) {
+		this(builder);
+		this.keySpace = keySpace;
+	}
+
+	/**
+	 * Internally used to set the job ID after instantiation.
+	 *
+	 * @param id
+	 * @throws Exception
+	 */
+	public void setJobId(String id) throws Exception {
+		super.setJobId(id);
+		table += id;
+	}
+
+	/**
+	 * Generates the necessary tables to store information.
+	 *
+	 * @return
+	 * @throws Exception
+	 */
+	@Override
+	public void createResource() throws Exception {
+		cluster = builder.getCluster();
+		session = cluster.connect();
+
+		session.execute(String.format("CREATE KEYSPACE IF NOT EXISTS %s with replication={'class':'SimpleStrategy', 'replication_factor':3};", keySpace));
+		session.execute(String.format("CREATE TABLE IF NOT EXISTS %s.%s (sink_id text, sub_id int, checkpoint_id bigint, PRIMARY KEY (sink_id, sub_id));", keySpace, table));
+
+		try {
+			session.close();
+		} catch (Exception e) {
+			LOG.error("Error while closing session.", e);
+		}
+		try {
+			cluster.close();
+		} catch (Exception e) {
+			LOG.error("Error while closing cluster.", e);
+		}
+	}
+
+	@Override
+	public void open() throws Exception {
+		if (builder == null) {
+			throw new RuntimeException("No ClusterBuilder was set.");
+		}
+		cluster = builder.getCluster();
+		session = cluster.connect();
+
+		deleteStatement = session.prepare(String.format("DELETE FROM %s.%s where sink_id='%s' and sub_id=%d;", keySpace, table, operatorId, subtaskId));
+		updateStatement = session.prepare(String.format("UPDATE %s.%s set checkpoint_id=? where sink_id='%s' and sub_id=%d;", keySpace, table, operatorId, subtaskId));
+		selectStatement = session.prepare(String.format("SELECT checkpoint_id FROM %s.%s where sink_id='%s' and sub_id=%d;", keySpace, table, operatorId, subtaskId));
+
+		session.execute(String.format("INSERT INTO %s.%s (sink_id, sub_id, checkpoint_id) values ('%s', %d, " + -1 + ");", keySpace, table, operatorId, subtaskId));
+	}
+
+	@Override
+	public void close() throws Exception {
+		session.executeAsync(deleteStatement.bind());
+		try {
+			session.close();
+		} catch (Exception e) {
+			LOG.error("Error while closing session.", e);
+		}
+		try {
+			cluster.close();
+		} catch (Exception e) {
+			LOG.error("Error while closing cluster.", e);
+		}
+	}
+
+	@Override
+	public void commitCheckpoint(long checkpointID) {
+		session.execute(updateStatement.bind(checkpointID));
```

Comment:
What happens if the `execute` call fails? Can we check the `ResultSet` for the success of the operation? If the commit did not succeed, then I think we should throw an exception.